### PR TITLE
Feature/add event public page api

### PR DESCRIPTION
**Pull request by dmitriy-uvin**
https://trello.com/c/zFOMiaXM/115-add-sending-email-when-event-is-stored  Notification
https://trello.com/c/Dy9mtuuq/140-create-event-after-event-was-saved-and-owner-was-notified Event

Add notification when event is saved.
Add event broadcasting.

### CODE REVIEW

File: api/app/Actions/Event/AddEventAction.php [hunk 1]
```diff
@@ -28,5 +30,9 @@ public function execute(AddEventRequest $request): void
         $event->timezone = $request->getTimezone();
 
         $this->eventRepository->save($event);
+
+        $event->eventType->owner->notify(new EventCreatedNotification($event));
+
+        event(new EventCreated());
```
Comment (aLLeXUs) [inline api/app/Actions/Event/AddEventAction.php:36]: The better way is to fire the event, and then add a listener for it that will send the notification.

File: api/app/Notifications/EventCreatedNotification.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    /**
+     * Get the mail representation of the notification.
+     *
+     * @return \Illuminate\Notifications\Messages\MailMessage
+     */
+    public function toMail($notifiable)
+    {
+        return (new MailMessage())
```
Comment (aLLeXUs) [inline api/app/Notifications/EventCreatedNotification.php:50]: Mail content creation should be moved into a separate class (ex. `NewEventEmail`).

File: api/app/Events/EventCreated.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    /**
+     * Get the channels the event should broadcast on.
+     *
+     * @return \Illuminate\Broadcasting\Channel|array
+     */
+    public function broadcastOn()
+    {
+        return new Channel('public-events');
```
Comment (aLLeXUs) [inline api/app/Events/EventCreated.php:44]: It is not a public event. Only the owner must receive the notification.

Comment (dmitriy-uvin) [reply api/app/Events/EventCreated.php:44]: I think that if that event creates by unauthorized user, so it's need to be public.
Will read about it.

File: api/app/Notifications/EventCreatedNotification.php (hunk 2 above)
Comment (dmitriy-uvin) [reply api/app/Notifications/EventCreatedNotification.php:50]: Class or blade template?

File: api/app/Actions/Event/AddEventAction.php (hunk 1 above)
Comment (dmitriy-uvin) [reply api/app/Actions/Event/AddEventAction.php:36]: @aLLeXUs Firstly call the event, then depends on it, notify the owner?

File: api/app/Events/EventCreated.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    /**
+     * Get the channels the event should broadcast on.
+     *
+     * @return \Illuminate\Broadcasting\Channel|array
+     */
+    public function broadcastOn()
+    {
+        return new PrivateChannel('public-events');
```
Comment (aLLeXUs) [inline api/app/Events/EventCreated.php:47]: PrivateChannel will add the needed prefix by itself, so we do not need the prefix here.

File: api/app/Notifications/EventCreatedNotification.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            ->subject("New Event: {$this->event->invitee_name},
+                    {$this->event->start_date}, {$this->eventType->name}")
+            ->markdown('notifications.event_created', [
+                'event' => $this->event,
+                'eventType' => $this->eventType,
+                'user' => $this->user
+            ]);
```
Comment (aLLeXUs) [inline api/app/Notifications/EventCreatedNotification.php:56]: All this content should not be here. Please, create [Mailable](https://laravel.com/docs/7.x/mail#generating-mailables) class that will contain all these things.

File: api/app/Providers/EventServiceProvider.php [hunk 6]
```diff
@@ -18,6 +19,9 @@ class EventServiceProvider extends ServiceProvider
         Registered::class => [
             SendEmailVerificationNotification::class,
         ],
+        EventCreated::class => [
+            EventCreatedListener::class
```
Comment (aLLeXUs) [inline api/app/Providers/EventServiceProvider.php:23]: Cmon, we were discussing this. We are going to create separate listeners for each action (from my examples: `SendEventCreatedNotification`, `AddEventToCalendar`, and so on).

File: api/routes/channels.php [hunk 7]
```diff
@@ -16,3 +16,7 @@
 Broadcast::channel('App.User.{id}', function ($user, $id) {
     return (int) $user->id === (int) $id;
 });
+
+Broadcast::channel('public-events', function () {
```
Comment (aLLeXUs) [inline api/routes/channels.php:20]: The prefix should not be here.

File: api/app/Providers/EventServiceProvider.php (hunk 6 above)
Comment (dmitriy-uvin) [reply api/app/Providers/EventServiceProvider.php:23]: @aLLeXUs We can add here another listeners wich will call different actions, but they all will listen one common event

Comment (aLLeXUs) [reply api/app/Providers/EventServiceProvider.php:23]: @dmitriy-uvin yes, the other listeners will be added, but currently, we have `EventCreatedListener` that only sends the notification, so we should name it properly.

Comment (dmitriy-uvin) [reply api/app/Providers/EventServiceProvider.php:23]: @aLLeXUs oh, i understand, sure

Comment (aLLeXUs) [reply api/app/Providers/EventServiceProvider.php]: @dmitriy-uvin just `SendEventCreatedNotification`

Comment (dmitriy-uvin) [reply api/app/Providers/EventServiceProvider.php]: @aLLeXUs done

**Pull request merged**